sematic/db/models/factories.py

```diff
@@ -148,6 +149,8 @@ def make_artifact(
     """
     Create an Artifact model instance from a value and type.
     """
+    if is_union(type_):
+        type_ = _get_value_type(value, type_)
```

Comment:
Can you write a test for this in `test_factories.py`?

Reply:
Sure I will add the tests for this there.